### DISCUSSION TITLE
How to add units avaible in style manager?

### DISCUSSION THREAD

**Discussion by confettidc** [GrapesJS/grapesjs#3990] [Q&A]
Hi,

I would like to add options "rem", "em" to margin options, how to do so?

![圖片](https://user-images.githubusercontent.com/5524183/144561104-91307628-3aac-447d-8c02-1aeff235a150.png)

Thanks.

**Comment by ronaldohoch** [comment]
Hello, i think this can help you: https://github.com/artf/grapesjs/discussions/3907

**Reply by confettidc** [reply]
Hi,

Thanks for your help and it seems that I have to redefine everything for just adding a few options in style manager (or other else, e.g. override existing panels), @artf will it possible to have "update" feature based on id defined in the future so that we no need to copy 90% of code from gjs-preset-webpage and only change small stuffs?

Your help is much appreciated. Thanks.

**Reply by confettidc** [reply]
Hi,

Finally I can search something similar at issue #2454:
https://github.com/artf/grapesjs/issues/2454

But I try to apply something similar to "margin-top":
```
// By default configuration, grapesjs has 2 `border-radius` properties, 1 for all sides
// and one of `composite` type with 4 inputs, so `getProperty` will return 
// an array with 2 types. I'm selecting the second one as it's a bit more complex
// than the first one (a simple input)
const brProperty = sm.getProperty('layout', 'margin', 'margin-top')[1];
// // Now you have to update inner properties (`border-top-left-radius`, ...)
brProperty.get('properties').forEach(property => property.set('units', ['px', '%', 'vh', 'em', 'rem']));
```

It returned error "brProperty is undefined".

Where can I find all default property list?

Thanks.
